homeassistant/helpers/storage.py

```diff
@@ -146,9 +155,55 @@ async def _async_load_data(self):
             # and we don't want that to mess with what we're trying to store.
             data = deepcopy(data)
         else:
-            data = await self.hass.async_add_executor_job(
-                json_util.load_json, self.path
-            )
+            try:
+                data = await self.hass.async_add_executor_job(
+                    json_util.load_json, self.path
+                )
+            except HomeAssistantError as err:
+                if isinstance(err.__cause__, JSONDecodeError):
+                    # If we have a JSONDecodeError, it means the file is corrupt.
+                    # We can't recover from this, so we'll log an error, rename the file and
+                    # return None so that we can start with a clean slate which will
+                    # allow startup to continue so they can restore from a backup.
+                    isotime = dt_util.utcnow().isoformat()
+                    corrupt_postfix = f".corrupt.{isotime}"
+                    corrupt_path = f"{self.path}{corrupt_postfix}"
+                    await self.hass.async_add_executor_job(
+                        os.rename, self.path, corrupt_path
+                    )
+                    _LOGGER.error(
+                        "Unrecoverable error decoding storage %s at %s; "
+                        "This may indicate an unclean shutdown, invalid syntax "
+                        "from manual edits, or disk corruption; "
+                        "The corrupt file has been saved as %s; "
+                        "It is recommended to restore from backup: %s",
+                        self.key,
+                        self.path,
+                        corrupt_path,
+                        err,
+                    )
+                    from .issue_registry import (  # pylint: disable=import-outside-toplevel
+                        IssueSeverity,
+                        async_create_issue,
+                    )
+
+                    async_create_issue(
```

Comment:
We may want to set the issue_domain according to the storage key, something like:
```py
issue_domain = HOMEASSISTANT_DOMAIN
if len(store_domain := self.key.split(".",1)) > 1 and store_domain[0] != "core:
    issue_domain = store_domain[0]
```

But maybe this is not reliable, I'm not sure if we enforce this pattern consistently.

Reply:
96788287da9a2dc328c830884a9e07e2e536ad4a